internal/support/logging/logger/file/file.go

```diff
@@ -13,69 +13,92 @@ import (
 	"io"
 	"os"
 	"path/filepath"
+	"sync"
 
-	"github.com/edgexfoundry/go-mod-core-contracts/models"
-)
+	"github.com/edgexfoundry/edgex-go/internal/support/logging/filter"
 
-const (
-	rmFileSuffix string = ".tmp"
+	"github.com/edgexfoundry/go-mod-core-contracts/models"
 )
 
-type fileLog struct {
+type Logger struct {
+	mutex    sync.Mutex
 	filename string
-	out      io.WriteCloser
+	file     io.WriteCloser
 }
 
-func (fl *fileLog) closeSession() {
-	if fl.out != nil {
-		fl.out.Close()
+func NewLogger(filename string) *Logger {
+	return &Logger{
+		filename: filename,
 	}
 }
 
-func (fl *fileLog) add(le models.LogEntry) error {
-	if fl.out == nil {
-		var err error
+func (l *Logger) openSession() (err error) {
```

Comment:
This function never returns error. Is it on purpose?

Reply:
It actually does.  It uses a named return parameter.  Whatever the value of that parameter is when  return is called is what is actually returned to the caller.  Functions/methods with named parameters can -- but don't have to -- have explicit values specified as part of the `return` statement.

Reply:
Thanks, I missed that language feature.